tools/toolchains/xcode_configure/xcode_sdk_frameworks.bzl

```diff
@@ -0,0 +1,141 @@
+PLATFORM_TUPLES = [
+    ("AppleTVOS", "tvos"),
+    ("AppleTVSimulator", "tvos"),
+    ("MacOSX", "macos"),
+    ("WatchOS", "watchos"),
+    ("WatchSimulator", "watchos"),
+    ("iPhoneOS", "ios"),
+    ("iPhoneSimulator", "ios"),
+]
+
+def _find_all_frameworks(sdk_path):
+    frameworks_dir = sdk_path.get_child("System").get_child("Library").get_child("Frameworks")
+    if not frameworks_dir.exists:
+        # Skip things like DriverKit
+        return []
+    frameworks = frameworks_dir.readdir()
+    return [
+        f.basename.replace(".framework", "")
+        for f in frameworks
+        if f.basename.endswith(".framework") and not f.basename.startswith("_")
+    ]
+
+def _build_file_header():
+    return """load("@build_bazel_rules_swift//swift:swift.bzl", "swift_module_alias", "swift_c_module")
+
+package(default_visibility = ["//visibility:public"])
+        
+"""
+
+IMPORTS_FILE = "bazel_xcode_imports.swift"
+
+def _get_dependency_graph(repository_ctx, developer_dir, sdk_path, target_triple):
+    frameworks = _find_all_frameworks(sdk_path)
+    swift_lib_dir = sdk_path.get_child("usr").get_child("lib").get_child("swift")
+    swift_libs = swift_lib_dir.readdir()
+    frameworks.extend([s.basename.replace(".swiftmodule", "") for s in swift_libs if s.basename.endswith(".swiftmodule") and not s.basename.startswith("_")])
+
+    import_file_content = "".join(["import {}\n".format(f) for f in frameworks])
+    repository_ctx.file(IMPORTS_FILE, content = import_file_content)
+    resource_dir = "{}/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift".format(developer_dir)
+
+    repository_ctx.report_progress("Scanning deps for {}".format(sdk_path))
+    deps_result = repository_ctx.execute(
+        [
+            "swiftc",
+            "-scan-dependencies",
+            "-sdk",
+            sdk_path,
+            "-resource-dir",
+            resource_dir,
+            "-target",
+            target_triple,
+            IMPORTS_FILE,
+        ],
+    )
+    repository_ctx.delete(IMPORTS_FILE)
+    if deps_result.return_code != 0:
+        fail("Could not scan dependencies for {}\n{}".format(sdk_path, deps_result.stderr))
+    return json.decode(deps_result.stdout)
+
+C_MODULE_TMPL = """swift_c_module(
+    name = "{name}_c",
+    module_name = "{name}",
+    system_module_map = "{module_map_file}",
+    deps = [
+{deps}
+    ],
+)
+
+"""
+
+SWIFT_MODULE_TMPL = """swift_module_alias(
+    name = "{name}_swift",
+    deps = [
+{deps}
+    ]
+)
+
+"""
+
+def _sub_bazel_path_vars(path, sdk_path, developer_dir):
+    path = path.replace(str(sdk_path), "__BAZEL_XCODE_SDKROOT__")
+    path = path.replace(str(developer_dir), "__BAZEL_XCODE_DEVELOPER_DIR__")
+    return path
+
+def _rule_for_pkg(developer_dir, sdk_path, pkg):
+    module_path = pkg.get("modulePath", "")
+    if not module_path or IMPORTS_FILE in module_path:
```

Comment:
Should this be a `fail` ? It looks like it detects a module we didn't load in?

Reply:
This is just parsing out stuff in the `swiftc` outputs that doesn't contain useful info (or like every other item looks like ```{"clang": "moduleName"}```)

Reply:
oh, and the imports file itself, which might be helpful for creating an umbrella header, but I'll follow up on that later.